collect_app/src/main/java/org/odk/collect/android/activities/FormFillingActivity.java

```diff
@@ -581,16 +588,7 @@ private void setupViewModels(FormEntryViewModelFactory formEntryViewModelFactory
                 return;
             }
             ValidationResult validationResult = consumable.getValue();
-            if (validationResult instanceof FailedValidationResult failedValidationResult) {
-                String errorMessage = failedValidationResult.getCustomErrorMessage();
-                if (errorMessage == null) {
-                    errorMessage = getString(failedValidationResult.getDefaultErrorMessage());
-                }
-                getCurrentViewIfODKView().setErrorForQuestionWithIndex(failedValidationResult.getIndex(), errorMessage);
-                swipeHandler.setBeenSwiped(false);
-            } else if (validationResult instanceof SuccessValidationResult) {
-                SnackbarUtils.showLongSnackbar(findViewById(R.id.llParent), getString(org.odk.collect.strings.R.string.success_form_validation), findViewById(R.id.buttonholder));
-            }
+            handleValidationResult(validationResult);
```

Comment:
I think we can potentially revise this now so that instead of `#getValidationResult`, all failed validations go through `#getCurrentIndex` and then we just need to expose the "success" notifications (that come out of `#validate`). That would also make the logic in `FormFillingActivity` simpler as we would always know if we're dealing with a success or a failure (unlike now where we have to account for both in `handleValidationResult`).

I know you're away next week, so that can totally be a follow-up if it doesn't seem like there's time. For the moment, maybe just deprecate `#getValidationResult` with a comment so it's clear we want to change that.